libdnf/rpm/repo_sack.cpp

```diff
@@ -59,7 +59,11 @@ static void libsolv_repo_free(LibsolvRepo * libsolv_repo) {
 RepoWeakPtr RepoSack::new_repo_from_libsolv_testcase(const std::string & repoid, const std::string & path) {
     std::unique_ptr<std::FILE, decltype(&std::fclose)> testcase_file(solv_xfopen(path.c_str(), "r"), &std::fclose);
     if (!testcase_file) {
-        throw SystemError(EIO, "Unable to open libsolv testcase file", path);
+        try {
+            throw SystemError(errno ? errno : EIO, path);
+        } catch (...) {
+            std::throw_with_nested(RuntimeError("Unable to open libsolv testcase file"));
+        }
```

Comment:
What is the purpose of this `try` - `catch` construct? Surely we won't be using this every time we throw an exception?

Reply:
I suggest that SystemError initialization can end in a problem with allocating memory. Therefore  there is a fallback that doesn't require any allocation.

Reply:
The exception with a system error is caught and nested in a higher-level exception.
It looks ugly in this place. It will not be standard to throw an exception and then re-throw it nested inside another exception in the same function.

Reply:
Well, can you put back the simple exception throw then? Or what's the reason to do this?